src/Markdowser/ViewModels/MainWindowViewModel.cs

```diff
@@ -196,50 +199,57 @@ private static bool IsValidHttpUri(string uriString, [NotNullWhen(true)] out Uri
         return Uri.TryCreate(uriString, UriKind.Absolute, out uri) && (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps);
     }
 
-    private void FetchUrl()
+private void FetchUrl()
+{
+    if (IsBusy)
     {
-        if (IsBusy)
-        {
-            WindowNotificationManager.Show(new Notification("Busy", "The browser is currently busy.", NotificationType.Warning));
-            return;
-        }
+        WindowNotificationManager.Show(new Notification("Busy", "The browser is currently busy.", NotificationType.Warning));
+        return;
+    }
 
-        if (string.IsNullOrWhiteSpace(Url))
+    if (string.IsNullOrWhiteSpace(Url))
+    {
+        Content = DefaultContent;
+        Debug.WriteLine("URL is empty.");
+        CurrentTab.Header = "New Tab";
+        return;
+    }
+
+    if (!IsValidHttpUri(Url, out _))
+    {
+        if (Url.StartsWith("//"))
         {
-            Content = DefaultContent;
-            Debug.WriteLine("URL is empty.");
-            CurrentTab.Header = "New Tab";
-            return;
+            Url = $"https:{Url}";
         }
-
-        if (!IsValidHttpUri(Url, out _))
+        else
         {
-            if (Url.StartsWith("//"))
+            // Search with duckduckgo
+            try
             {
-                Url = $"https:{Url}";
+                Url = string.Format(Settings.Current.SearchEngineUrl, Uri.EscapeDataString(Url));
             }
-            else
+            catch (FormatException ex)
             {
-                // Search with duckduckgo
-                try
-                {
-                    Url = string.Format(Settings.Current.SearchEngineUrl, Uri.EscapeDataString(Url));
-                }
-                catch (FormatException ex)
-                {
-                    WindowNotificationManager.Show(new Notification("Invalid Search Engine URL", $"{ex.Message}", NotificationType.Error));
-                }
+                WindowNotificationManager.Show(new Notification("Invalid Search Engine URL", $"{ex.Message}", NotificationType.Error));
             }
         }
+    }
 
-        IsBusy = true;
-        Progress = 0;
+    IsBusy = true;
+    Progress = 0;
 
-        _ = Task.Run(async () =>
-        {
-            Debug.WriteLine("Fetching URL...");
+    _ = Task.Run(async () =>
+    {
+        Debug.WriteLine("Fetching URL...");
 
-            try
+        try
+        {
+            ContentViewModelBase cachedContent = cacheService.Get(Url);
+            if (cachedContent != null)
```

Comment:
Suggestion as mentioned above

 ```suggestion
             if (cachedContent != null && useCache)
```